java/vector/src/main/java/org/apache/arrow/vector/BitVector.java

```diff
@@ -344,6 +344,22 @@ public final void get(int index, NullableBitHolder holder) {
       holder.isSet = 1;
       holder.value = get(index);
     }
+
+    /**
+     * Get the number nulls, this correspond to the number of bits set to 0 in the vector
+     * @return the number of bits set to 0
+     */
+    @Override
+    public final int getNullCount() {
+      int count = 0;
+      for (int i = 0; i < allocationSizeInBytes; ++i) {
+        byte byteValue = data.getByte(i);
+        // Java uses two's complement binary representation, hence 11111111_b which is -1 when converted to Int
+        // will have 32bits set to 1. Masking the MSB and then adding it back solves the issue.
+        count += Integer.bitCount(byteValue & 0x7F) - (byteValue >> 7);
+      }
+      return (allocationSizeInBytes * 8) - count;
```

Comment:
allocationSizeinBytes * 8 is rounded up to the next byte and not the actual valueCount.
If you rebase you will see that BitVector now overloads the load method. You can set valueCount there from fieldNode.getLength().